### PR TITLE
PIN-4277 purpose process: Added parameter eserviceId to GET -> /purposes/riskAnalysis/version/{riskAnalysisVersion}

### PR DESCRIPTION
**Pull request by nttdata-rtorsoli**
Detach branch from tag 1.0.17

### CODE REVIEW

File: src/test/scala/it/pagopa/interop/purposeprocess/PurposeApiServiceSpec.scala [hunk 1]
```diff
@@ -2509,62 +2510,235 @@ class PurposeApiServiceSpec extends AnyWordSpecLike with SpecHelper with Scalate
...
+        .returns(Future.successful(SpecData.tenant.copy(id = producerId, kind = PersistentTenantKind.GSP.some)))
+
+      Get() ~> service.retrieveRiskAnalysisConfigurationByVersion(eServiceId.toString, "1.0") ~> check {
+        status shouldEqual StatusCodes.OK
+        responseAs[RiskAnalysisFormConfigResponse].version shouldEqual "1.0"
+      }
+    }
+    "succeed when Tenant kind is PA and eService is Deliver" in {
+
+      val producerId: UUID = UUID.randomUUID()
```
Comment (galales) [inline src/test/scala/it/pagopa/interop/purposeprocess/PurposeApiServiceSpec.scala]: let's split the IDs to test that the requested tenant is actually the consumer
```suggestion
      val producerId: UUID = UUID.randomUUID()
      val consumerId: UUID = UUID.randomUUID()
```

File: src/test/scala/it/pagopa/interop/purposeprocess/PurposeApiServiceSpec.scala [hunk 2]
```diff
@@ -2509,62 +2510,235 @@ class PurposeApiServiceSpec extends AnyWordSpecLike with SpecHelper with Scalate
...
+      val eServiceId: UUID = UUID.randomUUID()
 
       implicit val context: Seq[(String, String)] =
         Seq("bearer" -> bearerToken, USER_ROLES -> "admin", ORGANIZATION_ID_CLAIM -> producerId.toString)
```
Comment (galales) [inline src/test/scala/it/pagopa/interop/purposeprocess/PurposeApiServiceSpec.scala]:
```suggestion
        Seq("bearer" -> bearerToken, USER_ROLES -> "admin", ORGANIZATION_ID_CLAIM -> consumerId.toString)
```

File: src/test/scala/it/pagopa/interop/purposeprocess/PurposeApiServiceSpec.scala [hunk 3]
```diff
@@ -2509,62 +2510,235 @@ class PurposeApiServiceSpec extends AnyWordSpecLike with SpecHelper with Scalate
...
+        .getEServiceById(_: UUID)(_: ExecutionContext, _: ReadModelService))
+        .expects(eServiceId, *, *)
+        .once()
+        .returns(
+          Future.successful(SpecData.eService.copy(id = eServiceId, producerId = UUID.randomUUID(), mode = Deliver))
+        )
+
       (mockTenantManagementService
         .getTenantById(_: UUID)(_: ExecutionContext, _: ReadModelService))
         .expects(producerId, *, *)
```
Comment (galales) [inline src/test/scala/it/pagopa/interop/purposeprocess/PurposeApiServiceSpec.scala]:
```suggestion
        .expects(consumerId, *, *)
```

File: src/test/scala/it/pagopa/interop/purposeprocess/PurposeApiServiceSpec.scala [hunk 4]
```diff
@@ -2509,62 +2510,235 @@ class PurposeApiServiceSpec extends AnyWordSpecLike with SpecHelper with Scalate
...
```
Comment (galales) [inline src/test/scala/it/pagopa/interop/purposeprocess/PurposeApiServiceSpec.scala]: same
```suggestion
        .expects(consumerId, *, *)
```

File: src/test/scala/it/pagopa/interop/purposeprocess/PurposeApiServiceSpec.scala [hunk 5]
```diff
@@ -2509,62 +2510,235 @@ class PurposeApiServiceSpec extends AnyWordSpecLike with SpecHelper with Scalate
...
```
Comment (galales) [inline src/test/scala/it/pagopa/interop/purposeprocess/PurposeApiServiceSpec.scala]: same
```suggestion
        .expects(consumerId, *, *)
```

File: src/test/scala/it/pagopa/interop/purposeprocess/PurposeApiServiceSpec.scala [hunk 6]
```diff
@@ -2509,62 +2510,235 @@ class PurposeApiServiceSpec extends AnyWordSpecLike with SpecHelper with Scalate
...
+        Seq("bearer" -> bearerToken, USER_ROLES -> "admin", ORGANIZATION_ID_CLAIM -> UUID.randomUUID().toString)
+
+      (mockCatalogManagementService
+        .getEServiceById(_: UUID)(_: ExecutionContext, _: ReadModelService))
+        .expects(eServiceId, *, *)
+        .once()
+        .returns(Future.failed(EServiceNotFound(eServiceId)))
+
+      Get() ~> service.retrieveRiskAnalysisConfigurationByVersion(eServiceId.toString, "1.0") ~> check {
+        status shouldEqual StatusCodes.InternalServerError
```
Comment (galales) [inline src/test/scala/it/pagopa/interop/purposeprocess/PurposeApiServiceSpec.scala]: being the eserviceId is a request parameter, IMO a 404 is better here
```suggestion
        status shouldEqual StatusCodes.NotFound
```

**Review by galales — COMMENTED** [review]
LGTM, just a couple of considerations

**Review by beetlecrunch — APPROVED** [review]
I approve, but I agree with @galales suggestions

File: src/test/scala/it/pagopa/interop/purposeprocess/PurposeApiServiceSpec.scala [hunk 7]
```diff
@@ -2512,7 +2512,7 @@ class PurposeApiServiceSpec extends AnyWordSpecLike with SpecHelper with Scalate
   "Purpose Risk Analysis Configuration for retrieving a specified version " should {
     "succeed when Tenant kind is PA and eService is Receive" in {
 
-      val producerId: UUID = UUID.randomUUID()
+      val consumerId: UUID = UUID.randomUUID()
```
Comment (galales) [inline src/test/scala/it/pagopa/interop/purposeprocess/PurposeApiServiceSpec.scala]: this was right
```suggestion
      val producerId: UUID = UUID.randomUUID()
```

File: src/test/scala/it/pagopa/interop/purposeprocess/PurposeApiServiceSpec.scala [hunk 8]
```diff
@@ -2538,7 +2538,7 @@ class PurposeApiServiceSpec extends AnyWordSpecLike with SpecHelper with Scalate
 
     "succeed when Tenant kind is PRIVATE and eService is Receive" in {
 
-      val producerId: UUID = UUID.randomUUID()
+      val consumerId: UUID = UUID.randomUUID()
```
Comment (galales) [inline src/test/scala/it/pagopa/interop/purposeprocess/PurposeApiServiceSpec.scala]: this was right
```suggestion
      val producerId: UUID = UUID.randomUUID()
```

File: src/test/scala/it/pagopa/interop/purposeprocess/PurposeApiServiceSpec.scala [hunk 9]
```diff
@@ -2564,7 +2564,7 @@ class PurposeApiServiceSpec extends AnyWordSpecLike with SpecHelper with Scalate
 
     "succeed when Tenant kind is GSP and eService is Receive" in {
 
-      val producerId: UUID = UUID.randomUUID()
+      val consumerId: UUID = UUID.randomUUID()
```
Comment (galales) [inline src/test/scala/it/pagopa/interop/purposeprocess/PurposeApiServiceSpec.scala]: this was right
```suggestion
      val producerId: UUID = UUID.randomUUID()
```

File: src/test/scala/it/pagopa/interop/purposeprocess/PurposeApiServiceSpec.scala [hunk 10]
```diff
@@ -2633,7 +2634,7 @@ class PurposeApiServiceSpec extends AnyWordSpecLike with SpecHelper with Scalate
 
       (mockTenantManagementService
         .getTenantById(_: UUID)(_: ExecutionContext, _: ReadModelService))
-        .expects(producerId, *, *)
+        .expects(consumerId, *, *)
         .once()
         .returns(Future.successful(SpecData.tenant.copy(id = producerId, kind = PersistentTenantKind.PRIVATE.some)))
```
Comment (galales) [inline src/test/scala/it/pagopa/interop/purposeprocess/PurposeApiServiceSpec.scala]:
```suggestion
        .returns(Future.successful(SpecData.tenant.copy(id = consumerId, kind = PersistentTenantKind.PRIVATE.some)))
```

**Review by galales — APPROVED** [review]

**Pull request merged**